**Problems for Quantum Materials Chemistry – Due Friday, March 13th at 4:00 pm**
A 1-time 10 point deduction will be made for problem sets submitted after 4:00 pm on Friday, March 8th.  

Each question is worth 20 points.

You are expected to adhere to the collaboration policy described in the syllabus.  As such, if you discuss strategies for solving problems with other students, you must declare this in the problem set solution you submit.  Sharing solutions with other students does not constitute acceptable collaboration, even if declared, and will be treated as academic dishonesty.


In [5]:
#Global inputs for Problem set 1
import numpy as np
pi = np.pi 
from matplotlib import pyplot as plt
import scipy as sp


1.  Consider an electron in a 1-D box of length $L=1$ nm that has energy eigenstates given by 
\begin{equation}
\psi_n(x) = \sqrt{2} {\rm sin}(n \pi x)
\end{equation}
when $x$ has units of nanometers.  Now consider that an atomically-precise hammer is used to deform the bottom of the box such that the electron feels an effective potential given by 
\begin{equation}
V(x) = V(x)= -2(x-0.5)^3+0.5.
\end{equation}

    a.  Compute the energy expectation value of an electron described by $\psi_1(x)$ after the box is deformed.

    b.  Explain if you think the original energy eigenstates, $\psi_n(x)$, are stationary 
states in the deformed box.  

In [6]:
# Code For Question 1

#Global Variables


x = np.linspace(0,1,100)




psi = np.sqrt(2)*np.sin(np.pi*x)

H_on_psi = ((np.pi**2*x**2)/(2)-2*(x-0.5)**3+0.5)



E_val = np.trapz(psi*H_on_psi,x)



print(E_val)
print('Energy expectation value is',E_val)
#plt.plot(x,V_x)






1.771059119365836
Energy expectation value is 1.771059119365836


Question 1
    a.  Compute the energy exxpectation value of an electron described by $\psi_1(x)$ after the box is deformed.
    b.  Explain if you think the original energy eigenstates, $\psi_n(x)$, are stationary 
states in the deformed box.

a.)Energy expectation value is -9158.861665137427
    
b.) This is a stationary state as none of the observables are Time dependent. the observables in this problem are position dependant for example the potential will effect the electron base on its spacial coordinates from the potential well.

2.  Considering the modified box in Question 1, write down the Hamiltonian operator after the box has been modified.  Using this Hamiltonian, use the linear variational method and a bsis of the first 3 energy eigenstates of the ordinary particle in a box to estimate the ground state energy of the modified system.

\begin{equation}
\hat{H} = \frac{\hbar^2\pi^2j^2}{2m}\delta_{ij}  
\end{equation}

In [18]:
def H_ij(i, j):

    ham_int = 0
    i = i+1
    j = j+1
    if i==j:
        ham_int = (np.pi**2 * j**2)/(2 * 1**2) + 2*np.sin(i*np.pi)*np.sin(j*np.pi) *-2*(i*j-0.5)**3+0.5 
    else:
        ham_int = (2)*np.sin((i*np.pi))*np.sin((j*np.pi))
 
    return ham_int



H_mat = np.zeros((3,3))
#print(H_mat)


for i in range(1,4):
    for j in range(1,4):
        H_mat[i-1][j-1] = H_ij(i, j)
#print(H_mat)



c = np.zeros(3)
c[0] = 1
Hc = np.dot(H_mat,c)
E = np.dot(np.transpose(c),Hc)
print(E)
#print(c)
#print(Hc)
E_opt, c_opt = np.linalg.eig(H_mat)
print('Energy Eigen value is',np.min(E_opt))







20.239208802178716
Energy Eigen value is 20.239208802178716
0.19739208802178715


a.) The Hamiltonian becomes.
\begin{equation}
\hat{H} = -\frac{\hbar^2}{2} \frac{d^2}{dx^2} - 2(x-0.5)^{3}+0.5.
\end{equation}


b.) Energy Eigen value is 20.239208802178716




3.  The predecessor to Hartree-Fock theory was the Hartree method, where the main difference is that the Hartree-Fock method includes anti-symmetry in the trial wavefunction by writing it as a Slater Determinant, while the Hartree method uses a simple product wavefunction that does not capture anti-symmetry.  In particular, for a minimal-basis model of H$_2$, the Hatree method's trial wavefunction is given by
\begin{equation}
\Psi_H = \phi_1(x_1) \phi_2(x_2)
\end{equation}
while the Hartree-Fock trial wavefunction is given by 
\begin{equation}
\Psi_{HF} = \phi_1(x_1) \phi_2(x_2) - \phi_1(x_2) \phi_2(x_1),
\end{equation}
where $\phi_1$ and $\phi_2$ are molecular orbitals, and $x_1$ and $x_2$ denote the coordinates
of electron 1 and electron 2, respectively.

    Write an expression for the total energy expectation values of $\Psi_H$ and $\Psi_{HF}$ in terms of 1- and 2-electron integrals and comment on the differences.  In particular, which method do you think will yield the lower total energy?  Recall the Hamiltonian operator can be written in terms of 1- and 2-electron operators as
\begin{equation}
\hat{H} = \sum_{i}^N \hat{h}_i + \sum_{i>j}^N \hat{V}_{i,j},
\end{equation}
    where $\hat{h}_i$ is the 1-electron operator for electron $i$ and $\hat{V}_{i,j}$ is the 2-electron operator for electrons $i$ and $j$.

hartree products produce energy expectation values lower that HF scince hatree product ignores electron correlation energy.(Energy expresion is written in file Foley_Ps1_Q3.pdf)

4. Use the following function as a trial wavefunction for the particle-in-a-box groundstate:
\begin{equation}
\phi(x) = c_1 x (L - x) + c_2 x (L-x)^2 + c_3 x^4(L-x) + c_4 x^2(L-x)^2.
\end{equation}
Compare the energy expectation value of $\phi(x)$ to $\psi_1(x)$, the true ground state of the particle in a box of length $L$.  Discuss whether or not this result is consistent with the Variational principle.

In [21]:
L = 1
x_ = np.linspace(0,5, 100)


def H_ij(i, j):

    ham_int = 0
    i = i+1
    j = j+1
    if i==j:
        ham_int = (np.pi**2 * j**2)/(2 * 1**2) +(j*(L-j))*(i*(L-i)) + (j*(L-j)**2)**2*(i*(L-i)**2)**2 + (j**4*(L-j))**2*(i**4*(L-i))**2 + (j**2*(L-j)**2)**2* (i**2*(L-i)**2)**2
    else:
        ham_int = (j*(L-j))*(i*(L-i)) + (j*(L-j)**2)**2*(i*(L-i)**2)**2 + (j**4*(L-j))**2*(i**4*(L-i))**2 + (j**2*(L-j)**2)**2* (i**2*(L-i)**2)**2
 
    return ham_int



H_mat = np.zeros((4,4))
#print(H_mat)


for i in range(1,5):
    for j in range(1,5):
        H_mat[i-1][j-1] = H_ij(i, j)
#print(H_mat)



c = np.zeros(4)

Hc = np.dot(H_mat,c)
E = np.dot(np.transpose(c),Hc)
E_opt, c_opt = np.linalg.eig(H_mat)
print(c_opt[0])
print('Energy Eigen value is',np.min(E_opt))

E_g = np.pi**2/(2*L**2)

print(E_g)


[-4.08176076e-05 -1.72537709e-03 -9.97703383e-01 -6.77124829e-02]
Energy Eigen value is 22.614876185884533
4.934802200544679


Setting the box length L to 1  the energy eigen value is 22.614876185884533 way larger than the groundstate value 4.934802200544679. The variational principle gives higher energy eigen values than the ground state so this is consistent with the variational principle.

5.  The energy expression for second-order Moller-Plesset Perturbation Theory is 
\begin{equation}
E_{MP2} = 2 \sum_{i,j,a,b}^N \frac{\langle ij | V | ab \rangle \langle ab | V | ij \rangle}{\epsilon_i + \epsilon_j - \epsilon_a - \epsilon_b} - \sum_{i,j,a,b}^N \frac{\langle ij | V | ab \rangle \langle ab | V | ji \rangle}{\epsilon_i + \epsilon_j - \epsilon_a - \epsilon_b}
\end{equation}
where $N$ is the number of molecular orbitals, $\epsilon_p$ is the molecular orbital energy
of orbital $p$, and $\langle pq | V | rs \rangle$ is the 2-electron integral
involving orbitals $\phi_p, \phi_q, \phi_r$, and $\phi_s$.  

**Explain how the computational cost of evaluating the MP2 energy scales with the number of molecular orbitals.**  

The compoutational cost with respect to molecular orbitals is $N^{4}$ and scince we are looking at four orbitals the computational cost will scale $4^{4}$ = 256